test/e2e/simple_vlan_and_ip_test.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("when vlan configured", func() {
+		var (
+			vlanId = "102"
+		)
+
+		BeforeEach(func() {
+			updateDesiredState(ifaceUpWithVlanUp(*firstSecondaryNic, vlanId))
+			waitForAvailableTestPolicy()
+		})
+		AfterEach(func() {
+			updateDesiredState(vlanAbsent(*firstSecondaryNic, vlanId))
+			resetDesiredStateForNodes()
+		})
+		It("should have the vlan interface configured", func() {
+			for _, node := range nodes {
+				vlanForNodeInterfaceEventually(node, fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)).Should(Equal(vlanId))
+			}
+		})
+	})
+	//TODO: change static IP to DHCP once we have a DHCP server running on a VLAN.
+	Context("when static address is configured on top of vlan interface", func() {
+		var (
+			ipAddresses = []string{"62.76.47.5", "62.76.47.6", "62.76.47.7"}
```

Comment:
We don't know the number of available nodes. You can at least build the IP based on the index of given node. Alternatively, setup the VLAN only on the first node.